lib/collection/src/collection/mod.rs

```diff
@@ -681,7 +681,7 @@ impl Collection {
                 continue;
             }
 
-            if this_peer_state != Some(Dead) || replica_set.is_dummy().await {
+            if !(this_peer_state == Some(Dead) || replica_set.is_dirty().await) {
```

Comment:
I wonder, we might always want to recover dummy shards here. In that case we can remove the dirty flag, and just use `is_dummy()`. If indeed possible, I'd prefer to remove it to keep our state simpler.

This might conflict with recovery mode though.

Let me have a bit of a thought about this.

Reply:
> This might conflict with recovery mode though.

Exactly why I avoided it.

Reply:
We should probably disable shard transfers all together when in recovery mode.

If we do that, we can simply check `is_dummy()` here without problems.

Reply:
I don't see checking a dirty flag as a stable way to see if recovery mode is enabled.

Instead we should explicitly check for recovery mode in the place where we drive transfers.

So I still prefer to:
- just check `is_dummy()` here
- explicitly check recovery mode